### PR TITLE
Update changelog for 21.06 release [skip ci]

### PR DESCRIPTION
**Pull request by sameerz**
Signed-off-by: Sameer Raheja <sraheja@nvidia.com>

Draft changelog PR for review.  Will update after final PR merge.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,182 @@
...
+|[#2330](https://github.com/NVIDIA/spark-rapids/issues/2330)|[FEA] Enable hash partitioning with arrays|
+|[#1103](https://github.com/NVIDIA/spark-rapids/issues/1103)|[FEA] Support date_format on GPU|
+|[#1125](https://github.com/NVIDIA/spark-rapids/issues/1125)|[FEA] explode() can take expressions that generate arrays|
+|[#1605](https://github.com/NVIDIA/spark-rapids/issues/1605)|[FEA] Support sorting on struct type keys|
+
+### Performance
+|||
+|:---|:---|
+|[#1445](https://github.com/NVIDIA/spark-rapids/issues/1445)|[FEA] GDS Integration|
+|[#2480](https://github.com/NVIDIA/spark-rapids/issues/2480)|[DOC] Split UCX docker examples into dockerfiles and add CentOS example|
```
Comment (tgravescs) [inline CHANGELOG.md:38]: this seems out of place in performance?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,182 @@
...
+|[#2436](https://github.com/NVIDIA/spark-rapids/issues/2436)|[FEA] Support nested types in CreateNamedStruct|
+|[#2461](https://github.com/NVIDIA/spark-rapids/issues/2461)|[FEA] Report gpuOpTime instead of totalTime for project, filter, window, limit|
+|[#2465](https://github.com/NVIDIA/spark-rapids/issues/2465)|[FEA] GpuFilterExec should report gpuOpTime not totalTime|
+|[#2013](https://github.com/NVIDIA/spark-rapids/issues/2013)|[FEA] Support concatenating ArrayType columns|
+|[#2425](https://github.com/NVIDIA/spark-rapids/issues/2425)|[FEA] Support for casting array of floats to array of doubles|
+|[#2012](https://github.com/NVIDIA/spark-rapids/issues/2012)|[FEA] Support Window functions(lead & lag) for ArrayType|
+|[#2011](https://github.com/NVIDIA/spark-rapids/issues/2011)|[FEA] Support creation of 2D array type|
+|[#1582](https://github.com/NVIDIA/spark-rapids/issues/1582)|[FEA] Allow StructType as input and output type to InMemoryTableScan and InMemoryRelation|
+|[#216](https://github.com/NVIDIA/spark-rapids/issues/216)|[FEA] Range window-functions must support non-timestamp order-by expressions|
```
Comment (tgravescs) [inline CHANGELOG.md:24]: I went ahead and closed https://github.com/NVIDIA/spark-rapids/issues/2483 from profiling tool if you want to add it into this list

**Review by revans2 — DISMISSED** [review]
Same comments as Tom. None of them are blockers, but we need to regenerate the changelogs again, so they are essentially the last thing to go in.

File: CHANGELOG.md (hunk 1 above)
Comment (sameerz) [reply CHANGELOG.md:38]: 2480 ended up listed as performance related because it had the `shuffle` label.  I removed the shuffle label.

**Comment by sameerz** [comment]
Updated the changelog just for review, I can update it one more time before we are ready to freeze.

**Review by revans2 — APPROVED** [review]

**Review by tgravescs — APPROVED** [review]

**Pull request merged**